File: cs_interface/write_serial.py
#!/usr/bin/python

import serial
import signal
import sys
import os

s = None

def exit_handler(signum, frame):
	global s
	print "exiting"	
	s.close()
	
def main(args, num_args):
	global s

	if num_args < 2:
		port = '/dev/pts/3'
	else:
		port = args[1]
	
	print 'opening port:', port
	s = serial.Serial()
	s.port = port 
	s.baudrate = 9600
	s.open()

	while(True):
		data = raw_input()
		data += '\n'
		s.write(data)

if __name__ == "__main__":
	signal.signal(signal.SIGINT, exit_handler)
	num_args = len(sys.argv)
	args = sys.argv
	main(args, num_args)

File: cs_interface/cpp/write_serial.cpp
#include <string.h>
#include <stdio.h>

#include "serial.h"


int main(int argc, char**argv)
{
	int i;
	char port[64] = "/dev/ttyUSB0";

	if(argc >= 2)
	{
		strcpy(port, argv[1]);
	}
	Serial s(port);
		

	printf("serial port opened: %s\n", port);


	for (i=0; i < 3; ++i)
	{
		s.swrite((char*)&i, 1);
	}


	return 0;
}

